api/v2/api.go

```diff
@@ -72,9 +74,17 @@ type API struct {
 type getAlertStatusFn func(prometheus_model.Fingerprint) types.AlertStatus
 
 // NewAPI returns a new Alertmanager API v2
-func NewAPI(alerts provider.Alerts, sf getAlertStatusFn, silences *silence.Silences, peer *cluster.Peer, l log.Logger) (*API, error) {
+func NewAPI(
+	alerts provider.Alerts,
+	marker types.Marker,
+	sf getAlertStatusFn,
```

Comment:
marker is no longer being passed in. check the new impl, it seems easier to understand to me that we pass everything in as a single `setAlertStatus` fn, instead of passing in one part (the `marker`) at creation time, and then passing in the updated-at-reload part separately in `Update`.